cpp/examples/symreg/symreg_example.cpp

```diff
@@ -0,0 +1,404 @@
+      if(!input_stream.is_open()){
+        std::cerr << "ERROR: Could not open input file " << train_weights << std::endl;
+        return 1;
+      }
+      
+      std::cout << "Reading input with " << n_train_rows << " rows"
+                << " from " << train_weights << "." << std::endl;
+      h_trainweights.reserve(n_train_rows);
+      while (input_stream >> val) {
+        h_trainweights.push_back(val);
+      }
+      input_stream.close();
+      input_stream.clear();
+    }
+    else{
+      h_trainweights.resize(n_train_rows,1.0f);
+    }
+
+    // Read test weights
+    if(test_weights != ""){
+      input_stream.open(test_weights,std::ios::in);
+      if(!input_stream.is_open()){
+        std::cerr << "ERROR: Could not open input file " << test_weights << std::endl;
+        return 1;
+      }
+      
+      std::cout << "Reading input with " << n_test_rows << " rows"
+                << " from " << test_weights << "." << std::endl;
+      h_testweights.reserve(n_test_rows);
+      while (input_stream >> val) {
+        h_testweights.push_back(val);
+      }
+      input_stream.close();
+      input_stream.clear();
+    }
+    else{
+      h_testweights.resize(n_test_rows,1.0f);
+    }
+  }
+  
+  // Check for valid mutation probability distribution
+  float p_sum = p_crossover + p_hoist + p_point + p_subtree;
+  if(p_sum >= 1.0f || p_sum <= 0.0f){
+    std::cerr << "ERROR: Invalid mutation probabilities provided" << std::endl 
+              << "Probability sum for crossover, subtree, host and point mutations is " << p_sum << std::endl;
+    return 1;
+  }
+
+  // Check if point_replace < 1.0f
+  if(p_point_replace > 1.0f || p_point_replace < 0.0f){
+    std::cerr << "ERROR: Invalid value for point replacement probability" << std::endl ;
+    return 1;
+  }
+
+  // Set all training parameters
+  params.num_features          = n_cols;
+  params.population_size       = population_size;
+  params.random_state          = random_state;
+  params.generations           = num_generations;
+  params.stopping_criteria     = stop_criterion;
+  params.p_crossover           = p_crossover;
+  params.p_subtree_mutation    = p_subtree;
+  params.p_hoist_mutation      = p_hoist;
+  params.p_point_mutation      = p_point;
+  params.p_point_replace       = p_point_replace;
+  params.parsimony_coefficient = parsimony_coeff;
+
+  // Set training metric
+  if(metric == "mae"){
+    params.metric = cg::metric_t::mae;
+  }else if(metric == "mse"){
+    params.metric = cg::metric_t::mse;
+  }else if(metric == "rmse"){
+    params.metric = cg::metric_t::rmse;
+  }else{
+    std::cerr << "ERROR: Invalid metric specified for regression (can only be mae, mse or rmse) " << std::endl;
+    return 1;
+  }
+
+  // Initialize and allocate device memory
+  std::cout << "***************************************" << std::endl;
+  std::cout << "Allocating device memory..." << std::endl;
+  raft::handle_t handle;
+  std::shared_ptr<ML::deviceAllocator> allocator(
```

Comment:
This seemed to give me error after I pulled the new branch-21.08. Changing to the following helped.
```suggestion
  std::shared_ptr<raft::mr::device::allocator> allocator(
```